storage/src/main/scala/coop/rchain/storage/package.scala

```diff
@@ -119,30 +129,39 @@ package object storage {
     * @param store
     * @param channel
     * @param data
+    * @param persist
     * @tparam C a type representing a channel
     * @tparam P a type representing a pattern
     * @tparam A a type representing a piece of data
     * @tparam K a type representing a continuation
     */
-  def produce[C, P, A, K](store: IStore[C, P, A, K], channel: C, data: A)(
+  def produce[C, P, A, K](store: IStore[C, P, A, K], channel: C, data: A, persist: Boolean)(
       implicit m: Match[P, A]): Option[(K, List[A])] =
     store.withTxn(store.createTxnWrite()) { txn =>
       val groupedChannels: List[List[C]] = store.getJoin(txn, channel)
-      store.putA(txn, List(channel), data)
+      store.putA(txn, List(channel), Datum(data, persist))
```

Comment:
This is going to make your life difficult, but the same logic needs to apply here.

What that means is that you can't put it in the store and then loop over the affected continuations
You'll have to pass the (data, channel) pair to extractProduceCandidates